### PR TITLE
Add dockerfile for  bittensor 

### PR DESCRIPTION
**Pull request by shibshib**
This PR adds the ability to dockerize bittensor, with a CUDA-compatible container so as to take full advantage of any GPUs on the host machine.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+RUN apt-get install -y -V libarrow-dev 
+RUN apt-get install -y -V libarrow-glib-dev 
+RUN apt-get install -y -V libarrow-dataset-dev 
+RUN apt-get install -y -V libarrow-flight-dev
+
+WORKDIR /bittensor
+RUN pip install --upgrade numpy pandas setuptools "tqdm>=4.27,<4.50.0"
+RUN pip install -r requirements.txt
+RUN pip install -e . 
+RUN ./scripts/create_wallet.sh
```
Comment (unconst) [inline Dockerfile]: Not sure we should create a wallet for the user, they wont have access to this.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+EXPOSE 8091
+VOLUME ["/bittensor"]
```
Comment (unconst) [inline Dockerfile]: Isn't the volume ~/.bittensor?

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+services:
+  dev:
+    container_name: node-bittensor
+    image: "bittensor/bittensor:latest"
+    ports:
+      - "8091:8091"
+    
+    command: /bin/bash -c "
+      cd /bittensor 
+      && python3.7 examples/IMAGE/mnist.py --metagraph.chain_endpoint 172.17.0.1:9944"
```
Comment (unconst) [inline docker-compose.yml]: Does this point to feynman?

File: requirements.txt [hunk 4]
```diff
@@ -40,4 +39,7 @@ torchvision
 transformers
 validators
 xxhash==1.3.0
-codecov
\ No newline at end of file
+pandas
```
Comment (unconst) [inline requirements.txt:42]: Did you say this worked because we install cython first?
I thought you mentioned that this doesnt work?

**Review by unconst — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (shibshib) [reply Dockerfile]: It doesn't work without a wallet though, so it needs some sort of wallet. Unless we assume users have a wallet in their ~/.bittensor?

File: Dockerfile (hunk 2 above)
Comment (shibshib) [reply Dockerfile]: No, this is the bittensor code volume, which will be mounted in the root dir of the image

File: docker-compose.yml (hunk 3 above)
Comment (shibshib) [reply docker-compose.yml]: No, this is the docker bridge network between the two containers running on the same computer.

File: requirements.txt (hunk 4 above)
Comment (shibshib) [reply requirements.txt:42]: We need to install pandas, it wasn't added anywhere, so not sure what you mean :)

Comment (shibshib) [reply requirements.txt:42]: Ah, just caught what you meant. Yes I moved it to the bottom before thinking that the install order mattered, but it apparently doesn't.

File: Dockerfile (hunk 1 above)
Comment (shibshib) [reply Dockerfile]: We can always ensure users created their wallet before running this, so that it copies it over to the container

Comment (unconst) [reply Dockerfile]: really think this should be taken out
if a user runs they will be subscribing with these keys.
Not good.

File: Dockerfile (hunk 2 above)
Comment (unconst) [reply Dockerfile]: right, i think we can mount the root dir so we can access the wallet created by the user

**Review by unconst — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (shibshib) [reply Dockerfile]: Done!

File: Dockerfile (hunk 2 above)
Comment (shibshib) [reply Dockerfile]: Good idea, done!

**Review by unconst — APPROVED** [review]

**Review by krishadhithya — COMMENTED** [review]
LGTM @unconst

**Pull request merged**